daemon/cluster/executor/container/controller.go

```diff
@@ -217,17 +217,35 @@ func (r *controller) Start(ctx context.Context) error {
 
 				continue
 			}
-
+			// We should only remove networks after container networks are initialized.
+			// However, there are too many possible errors during container starting.
+			// Since removeNeworks() is an idempotent operation, it should be OK to removeNetworks
+			// here although it's not ideal.
+			log.G(ctx).Debugf("Removing networks after container %s failed to start.", r.adapter.container.name())
+			r.adapter.removeNetworks(ctx)
```

Comment:
Should errors that occur during removal of the network be handled? (perhaps only logged if we can't / don't have to handle them otherwise)

Reply:
Yes, we need log the error of `removeNetworks(ctx)`, but return the error indicating the reason for container-start.